data-service/src/main/java/com/amazonaws/blox/dataservice/storage/DeploymentDDBStore.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.dataservice.storage;
+
+import com.amazonaws.blox.dataservice.model.Deployment;
+import com.amazonaws.blox.dataservice.model.DeploymentStatus;
+import com.amazonaws.blox.dataservice.storage.model.DeploymentDDBRecord;
+import com.amazonaws.blox.dataservice.storage.model.DeploymentDDBRecordTranslator;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBScanExpression;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@AllArgsConstructor
+public class DeploymentDDBStore implements DeploymentStore {
+
+  @NonNull private DynamoDBMapper dynamoDBMapper;
+
+  @Override
+  public Deployment createDeployment(final Deployment deployment) {
+    final DeploymentDDBRecord record =
+        DeploymentDDBRecordTranslator.toDeploymentDDBRecord(deployment);
+    dynamoDBMapper.save(record);
+    return DeploymentDDBRecordTranslator.fromDeploymentDDBRecord(record);
+  }
+
+  @Override
+  public Deployment updateDeployment(final Deployment deployment) {
+    DeploymentDDBRecord record =
+        dynamoDBMapper.load(DeploymentDDBRecord.withHashKey(deployment.getDeploymentId()));
```

Comment:
What does "higher chance of success" mean here? It kind of looks like we're just ignoring conflicts: in particular, it looks like the RecordTranslator [doesn't update the `version` field on the new record to match the one that was given here](https://github.com/blox/blox/pull/231/files#diff-5a38996365ad856775ba66b351b16b0fR36), so if deployment status was updated after this code initially read the record, then the DDB Mapper's optimistic concurrency won't catch it. Is that the intention? If not, this approach should be doing some of its own checking for dirty fields, and failing early if anything might conflict.

Additionally, it's entirely opaque to the consumer of this API that the only fields that will get updated from the given Deployment are `deploymentStatus` and the timestamps.

Is the rest of the record considered immutable? If so, a dedicated `updateStatus(DeploymentId, Status)` API might be less confusing, and provide a better boundary for concurrency control.